src/CounterInterceptor/ClientCounter.cs

```diff
@@ -0,0 +1,89 @@
+namespace TemporalioSamples.CounterInterceptor;
+
+using System.Numerics;
+
+public class ClientCounter
+{
+    private const string NumberOfWorkflowExecutions = "numOfWorkflowExec";
+    private const string NumberOfSignals = "numOfSignals";
+    private const string NumberOfQueries = "numOfQueries";
+    private static Dictionary<string, Dictionary<string, BigInteger?>> perWorkflowIdDictionary =
```

Comment:
```suggestion
    private static Dictionary<string, Dictionary<string, uint64>> perWorkflowIdDictionary =
```

I think `uint64` is enough. If you incremented the counter every nanosecond (not very possible), it'd take you over 580 years to overflow.

Reply:
Changed to uint. Had style issues with uint64